internal/bminventory/inventory.go

```diff
@@ -3798,14 +3798,12 @@ func (b *bareMetalInventory) DownloadMinimalInitrd(ctx context.Context, params i
 	var netFiles []staticnetworkconfig.StaticNetworkConfigData
 	var scriptContent, serviceContent string
 	if infraEnv.StaticNetworkConfig != "" {
-		// backward compatibility - nmstate.service has been available on RHCOS since version 4.14+, therefore, we should maintain both flows
-		var ok bool
-		ok, err = b.staticNetworkConfig.NMStatectlServiceSupported(infraEnv.OpenshiftVersion)
+		var shouldUseNmstateService bool
```

Comment:
Seems that we're missing tests for both flows in inventory_test. Can be handled later on in a separate PR.